### PR TITLE
Feature/terraform

### PR DESCRIPTION
**Pull request by trottomv**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,39 +1,151 @@
 .DEFAULT_GOAL := help
 
+.PHONY: behave
```
Comment (trottomv) [inline Makefile:3]: verify the reals changes of this file

File: requirements.in [hunk 2]
```diff
@@ -1,12 +0,0 @@
-# pip-compile --output-file requirements.txt requirements.in
```
Comment (trottomv) [inline requirements.in:1]: maybe this requirements.in are still necessary? verify pls

File: requirements.txt [hunk 3]
```diff
@@ -1,104 +0,0 @@
-#
```
Comment (trottomv) [inline requirements.txt:1]: maybe this requirements.txt is still necessary? verify pls

File: requirements/base.in [hunk 4]
```diff
@@ -0,0 +1 @@
+# pip-compile --output-file base.txt base.in
```
Comment (trottomv) [inline requirements/base.in:1]: maybe these requirements/*.in and requirements/*.txt are not necessary? verify pls

File: {{cookiecutter.project_dirname}}/.env_template [hunk 5]
```diff
@@ -1,16 +1,17 @@
...
-DJANGO_ADMINS=admin,errors@{{cookiecutter.domain_url}}
-DJANGO_ALLOWED_HOSTS=localhost
+DATABASE_URL=postgres://postgres:postgres@postgres:5432/{{ cookiecutter.project_slug }}
+DJANGO_ADMINS=admin,errors@example.org
+DJANGO_ALLOWED_HOSTS=localhost,{{ cookiecutter.service_slug }}
+DJANGO_BUILD_TARGET=local
 DJANGO_CONFIGURATION=Local
 DJANGO_DEBUG=True
-DJANGO_DEFAULT_FROM_EMAIL=info@{{cookiecutter.domain_url}}
+DJANGO_DEFAULT_FROM_EMAIL=info@example.org
```
Comment (trottomv) [inline {{cookiecutter.project_dirname}}/.env_template:9]: the placeholder {{ cookiecutter.domain_url }} could still be useful.

File: {{cookiecutter.project_dirname}}/.env_template [hunk 6]
```diff
@@ -1,16 +1,17 @@
...
 DJANGO_SECRET_KEY=__SECRETKEY__
-DJANGO_SERVER_EMAIL=server@{{cookiecutter.domain_url}}
+DJANGO_SERVER_EMAIL=server@example.org
 DJANGO_SESSION_COOKIE_DOMAIN=localhost
-DJANGO_SUPERUSER_EMAIL=info@{{cookiecutter.domain_url}}
-DJANGO_SUPERUSER_PASSWORD={{cookiecutter.project_slug}}
+DJANGO_SUPERUSER_EMAIL=info@example.org
```
Comment (trottomv) [inline {{cookiecutter.project_dirname}}/.env_template:13]: the placeholder {{ cookiecutter.domain_url }} could still be useful.

File: Makefile [hunk 7]
```diff
@@ -20,9 +24,17 @@ local: pip_update  ## Install local requirements and dependencies
...
+	python3 manage.py permissions --path artifacts/permissions.csv
+
 .PHONY: pip
 pip: pip_update  ## Compile requirements
-	python3 -m piptools compile --no-header --quiet --upgrade --output-file requirements.txt requirements.in
+	python3 -m piptools compile --no-header --quiet --upgrade --output-file requirements/base.txt requirements/base.in
+	python3 -m piptools compile --no-header --quiet --upgrade --output-file requirements/common.txt requirements/common.in
+	python3 -m piptools compile --no-header --quiet --upgrade --output-file requirements/local.txt requirements/local.in
+	python3 -m piptools compile --no-header --quiet --upgrade --output-file requirements/remote.txt requirements/remote.in
+	python3 -m piptools compile --no-header --quiet --upgrade --output-file requirements/test.txt requirements/test.in
```
Comment (filippo-20tab) [inline Makefile:37]:
```suggestion
	python3 -m piptools compile --no-header --quiet --upgrade --output-file requirements/common.txt requirements/common.in
	python3 -m piptools compile --no-header --quiet --upgrade --output-file requirements/local.txt requirements/local.in
	python3 -m piptools compile --no-header --quiet --upgrade --output-file requirements/test.txt requirements/test.in
```
We agreed with @pauloxnet to only user 3 levels.

File: cookiecutter.json [hunk 8]
```diff
@@ -1,9 +1,13 @@
 {
   "project_name": null,
   "project_slug": "{{ cookiecutter.project_name | slugify(separator='') }}",
-  "project_dirname": "{{ cookiecutter.project_name | slugify(separator='') }}",
-  "domain_url": "{{ cookiecutter.project_name | slugify(separator='') }}.com",
-  "gitlab_group_slug": "{{ cookiecutter.project_name | slugify(separator='') }}",
-  "use_media": ["Yes", "No"],
+  "service_slug": "{{ cookiecutter.project_slug }}",
+  "project_dirname": ["{{ cookiecutter.service_slug }}", "{{ cookiecutter.project_slug }}"],
```
Comment (filippo-20tab) [inline cookiecutter.json:5]:
```suggestion
  "project_dirname": ["{{ cookiecutter.service_slug | slugify(separator='') }}", "{{ cookiecutter.project_slug }}"],
```

File: cookiecutter.json [hunk 9]
```diff
@@ -1,9 +1,13 @@
...
+  "django_settings_dirname": "{{ cookiecutter.project_slug | slugify(separator='') }}",
+  "internal_service_port": "8000",
+  "project_url_dev": null,
+  "project_url_stage": null,
+  "project_url_prod": null,
+  "media_storage": ["local", "s3-digitalocean", "none"],
```
Comment (filippo-20tab) [inline cookiecutter.json:11]:
```suggestion
  "media_storage": ["local", "s3", "none"],
```

File: requirements/common.in [hunk 10]
```diff
@@ -0,0 +1,9 @@
+# pip-compile --output-file common.txt common.in
+-r base.in
+
```
Comment (filippo-20tab) [inline requirements/common.in:3]:
```suggestion
```

File: cookiecutter.json (hunk 9 above)
Comment (filippo-20tab) [inline cookiecutter.json:11]:
```suggestion
  "media_storage": ["local", "s3", "none"],
```

File: requirements/common.in [hunk 11]
```diff
@@ -0,0 +1,9 @@
...
+black~=22.1.0
```
Comment (filippo-20tab) [inline requirements/common.in:4]:
```suggestion
black~=22.1.0
```

File: requirements/local.in [hunk 12]
```diff
@@ -0,0 +1,7 @@
+# pip-compile --output-file local.txt local.in
+-r test.in
+
+ipdb~=0.13.0
+pip-tools~=6.5.0
+pre-commit~=2.17.0
```
Comment (filippo-20tab) [inline requirements/local.in:6]:
```suggestion
pre-commit~=2.17.0
```

File: requirements/test.in [hunk 13]
```diff
@@ -0,0 +1,11 @@
+# pip-compile --output-file test.txt test.in
+-r common.in
+
+black~=22.1.0
```
Comment (filippo-20tab) [inline requirements/test.in:4]:
```suggestion
```

File: {{cookiecutter.project_dirname}}/.env_template (hunk 5 above)
Comment (filippo-20tab) [reply {{cookiecutter.project_dirname}}/.env_template:10]: No need for that in local development.

File: {{cookiecutter.project_dirname}}/{{cookiecutter.django_settings_dirname}}/workers.py [hunk 14]
```diff
@@ -1,4 +1,4 @@
-"""Custom uvicorn workers."""
+"""Define gunicorn supported workers."""
```
Comment (filippo-20tab) [inline {{cookiecutter.project_dirname}}/{{cookiecutter.django_settings_dirname}}/workers.py:1]:
```suggestion
"""Custom gunicorn supported worker."""
```

**Pull request merged**